[ci-skip][Docs]Add sqlite3_production_warning to configuring guide

hachi8833 opened the pull request:
### Motivation / Background

This Pull Request has been created because the `config.active_record.sqlite3_production_warning` config that was merged as https://github.com/rails/rails/pull/42191 in [Rails 7.0.0.alpha1](https://github.com/rails/rails/blob/7-0-stable/activerecord/CHANGELOG.md#rails-700alpha1-september-15-2021) still has not been added to the configuring guide.

### Detail

This Pull Request adds the entry for `config.active_record.sqlite3_production_warning`.

### Additional information

As far as I see, the current config entries for Active Record are unsorted and I'm a bit wondering if the line I added is appropriate or not. This time I added it to the next line of `sqlite3_adapter_strict_strings_by_default`.

Or if you think adding `config.active_record.sqlite3_production_warning` to the configuring guide is redundant because the warning indicates the one, just close the PR.

```
Oct 14 05:38:10 PM  W, [2023-10-14T08:38:07.617001 #1]  WARN -- : You are running SQLite in production, this is generally not recommended. You can disable this warning by setting "config.active_record.sqlite3_production_warning=false".
```

cc: @intrip

### Checklist

Before submitting the PR make sure the following are checked:

* [x] This Pull Request is related to one change. Changes that are unrelated should be opened in separate PRs.
* [x] Commit message has a detailed description of what changed and why. If this PR fixes a related issue include it in the commit message. Ex: `[Fix #issue-number]`

jonathanhefner committed:
Support Active Model attribute type decoration

This adds a `decorate_attributes` method to Active Model to support
attribute type decoration.  `decorate_attributes` is a private,
low-level API that is intended to be wrapped by high-level APIs like
`ActiveRecord::Base::normalizes` and `ActiveRecord::Base::enum`.

jonathanhefner committed:
Use ActiveModel::AttributeRegistration in AR

This refactors the `ActiveRecord::Attributes` module to use
`ActiveModel::AttributeRegistration`.  This also replaces the block form
of the `attribute` method (which was support by only Active Record) with
`decorate_attributes` (which is supported by both Active Model and
Active Record).  The block form of the `attribute` method was a private
API, so no deprecation is necessary.

stevepolitodesign committed:
Improve generator logging and debugging

This commit is a follow-up to #49448, with regards to [this comment][].

Prior to this commit, logging and debugging for
[Rails::Generators::Actions][] was limited even when enabling
`RAILS_LOG_TO_STDOUT`. This was because the private `action` method in
the corresponding test was capturing `$stdout`.

[this comment]: https://github.com/rails/rails/pull/49448#issuecomment-1743061538
[Rails::Generators::Actions]: https://api.rubyonrails.org/v7.0.8/classes/Rails/Generators/Actions.html

zzak committed:
Revert "Freeze rack in 2.0 for guides bug report templates"

This reverts commit c5ace249e742eeb5f5aee4b4464b9c6f7a4af3eb.

Since this commit support for Rack 3 has stabilized.

hachi8833 committed:
Add sqlite3_production_warning to configuring guide

rafaelmontas committed:
Fix typo in configuring guide

fxn committed:
Do not memoize auto/eager load paths in engines

dustinbrownman committed:
Fix code blocks that caused lint failures

zzak commented:
Should this be added to the default values as well?

https://github.com/rails/rails/blob/94ff1bf0dcc38cc9176844d90ecfa54a5a09dada/guides/source/configuring.md?plain=1#L92

zzak committed:
Suppress mail warning for "unused variable - disp_type_s"

e.g.:

```
zzak@mbp16 railties % bin/test test/application/assets_test.rb
Run options: --seed 32028

.............../Users/zzak/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/mail-2.8.1/lib/mail/parsers/content_location_parser.rb:592: warning: assigned but unused variable - disp_type_s
I, [2023-10-15T10:38:33.005925 #97662]  INFO -- : [0ca46786-853f-4740-9e71-f644c4893502] Started GET "/assets/demo.js" for 127.0.0.1 at 2023-10-15 10:38:33 +0900
E, [2023-10-15T10:38:33.006641 #97662] ERROR -- : [0ca46786-853f-4740-9e71-f644c4893502]
[0ca46786-853f-4740-9e71-f644c4893502] ActionController::RoutingError (No route matches [GET] "/assets/demo.js"):
[0ca46786-853f-4740-9e71-f644c4893502]
../Users/zzak/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/mail-2.8.1/lib/mail/parsers/content_location_parser.rb:592: warning: assigned but unused variable - disp_type_s
..../Users/zzak/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/mail-2.8.1/lib/mail/parsers/content_location_parser.rb:592: warning: assigned but unused variable - disp_type_s
I, [2023-10-15T10:38:33.973335 #97660]  INFO -- : [fa848d82-0d80-4b23-925b-73fa5d0f47d4] Started GET "/posts?debug_assets=true" for 127.0.0.1 at 2023-10-15 10:38:33 +0900
I, [2023-10-15T10:38:33.975021 #97660]  INFO -- : [fa848d82-0d80-4b23-925b-73fa5d0f47d4] Processing by PostsController#index as HTML
I, [2023-10-15T10:38:33.975209 #97660]  INFO -- : [fa848d82-0d80-4b23-925b-73fa5d0f47d4]   Parameters: {"debug_assets"=>"true"}
I, [2023-10-15T10:38:33.977560 #97660]  INFO -- : [fa848d82-0d80-4b23-925b-73fa5d0f47d4] Completed 200 OK in 2ms (Views: 1.9ms | ActiveRecord: 0.0ms | Allocations: 1266)
......./Users/zzak/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/mail-2.8.1/lib/mail/parsers/content_location_parser.rb:592: warning: assigned but unused variable - disp_type_s
.

Finished in 4.842481s, 5.9887 runs/s, 23.3351 assertions/s.
29 runs, 113 assertions, 0 failures, 0 errors, 0 skips
```

For example:
https://buildkite.com/rails/rails/builds/94132#0186806f-8a23-4d07-ae8f-3b937f8517ae/1162-1171

See also #47484

hachi8833 commented:
@zzak That's what I want to know about.

Current CI makes us to add the entry to the new framework file as well if we add the entry to the list you specified. The config just control the warning and irrelevant to compatibility issuees, so I'm afraid It might be unnecessary to add. 🤔

I'd be glad if some core members would suggest about this.

zzak commented:
This is just my opinion, but not a very strong one, I could be wrong.

> Below are the default values associated with each target version. In cases of conflicting values, newer versions take precedence over older versions.

It doesn't explicitly say "all default values" for each version, but to me it looks like it could also be seen as a Table of Contents organized by version, where the contents below this are sorted by framework.

rafaelfranca committed:
We don't use `::` to denote class methods

rafaelfranca committed:
Avoid __method__

Just be explicit about what we are trying to do here.

rafaelfranca committed:
Whitespaces

rafaelfranca committed:
Add psych to the bug report template

With release of psych 5.1.1.1 bundler doesn't know which version to
load.

https://buildkite.com/rails/rails/builds/100852

rafaelfranca committed:
Revert "Add psych to the bug report template"

This reverts commit e4e242685efc74e4bacccd5c5878bc671fd63fb7.

This doesn't solve the problem.

eileencodes committed:
[ci skip] Fix shard docs

We never explained how migrations paths work for shards. This fixes that
and also adds the appropriate class setup. You no longer need to set a
`default` shard as of #48353. In addition, `ApplicationRecord` should be
used for the non-sharded db that also serves as the tenant/shard router.
Then shards should get their own connection class since the schema
differs.

eileencodes committed:
[ci skip] Fix shard docs followup

Oops, the ShardRecord needs to be an abstract class and
ApplicationRecord needs to be a primary_abstract_class

seanpdoyle committed:
Fix failing linter in `guides/source`

Fix [Failing CI Lint][]

Adds a missing comma to make an example Ruby block syntactically valid.

[Failing CI Lint]: https://github.com/rails/rails/actions/runs/6539678344/job/17758241844?pr=49486#step:4:12

yahonda committed:
Bump psych to 5.1.1.1

This commit addresses action_controller_gem.rb failure by updating psych as follows.
`$ bundle update psych --conservative`

Refer to https://buildkite.com/rails/rails/builds/100884#018b3a84-f1c5-4461-839c-5cfbe26ee001/1092-1125

- Without this commit
```
$ ruby -v
ruby 3.2.2 (2023-03-30 revision e51014f9c0) [x86_64-linux]
$ cd guides/bug_report_templates
$ ruby action_controller_gem.rb
Fetching gem metadata from https://rubygems.org/...........
... snip ...
Using rails 7.1.1
/home/yahonda/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/bundler-2.3.22/lib/bundler/runtime.rb:308:in `check_for_activated_spec!': You have already activated psych 5.1.0, but your Gemfile requires psych 5.1.1.1. Prepending `bundle exec` to your command may solve this. (Gem::LoadError)
	from /home/yahonda/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/bundler-2.3.22/lib/bundler/runtime.rb:25:in `block in setup'
	from /home/yahonda/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/bundler-2.3.22/lib/bundler/spec_set.rb:155:in `each'
	from /home/yahonda/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/bundler-2.3.22/lib/bundler/spec_set.rb:155:in `each'
	from /home/yahonda/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/bundler-2.3.22/lib/bundler/runtime.rb:24:in `map'
	from /home/yahonda/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/bundler-2.3.22/lib/bundler/runtime.rb:24:in `setup'
	from /home/yahonda/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/bundler-2.3.22/lib/bundler/inline.rb:66:in `block in gemfile'
	from /home/yahonda/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/bundler-2.3.22/lib/bundler/settings.rb:131:in `temporary'
	from /home/yahonda/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/bundler-2.3.22/lib/bundler/inline.rb:50:in `gemfile'
	from action_controller_gem.rb:5:in `<main>'
$
```

skipkayhil committed:
Fix extra blank line when no error_highlight

A new condition was recently [added][1] to exclude error_highlight from
the Gemfile when the Ruby version is 3.2 or greater. However, this
change leads to an extra blank line after `gem "spring"` on Ruby 3.2+.
This could also happen to new apps generated on Rubies < 3.1, but my
guess is that is less common.

This commit fixes the issue by moving the extra blank line into the
condition.

[1]: d7b395145c644434b1c2be07ef334ea5cb643b75

akhilgkrishnan committed:
Add description for error_highlight gem

aitor committed:
Update classic_to_zeitwerk_howto.md

Some typos were fixed on the Zeitwerk migration guide.

skipkayhil commented on guides/source/configuring.md:
```diff
@@ -1458,6 +1458,17 @@ The default value depends on the `config.load_defaults` target version:
+
+Specifies whether a warning should be raised when using SQLite3 in production environment.
+
+The default value depends on the `config.load_defaults` target version:
+
+| Starting with version | The default value is |
+| --------------------- | -------------------- |
+| (original)            | `false`              |
+| 7.0                   | `true`               |
+
```
```suggestion
The default value is `true`
```

skipkayhil reviewed (commented):
The first line of "Versioned Default Values" defines "default values" for a "target version" in reference to `load_defaults` specifically:
> `config.load_defaults` loads default configuration values for a target version...
> 
> Below are the default values associated with each target version...
> #### Default Values for Target Version ...

Since this configuration is not part of `load_defaults`, it should not include the table in the description, and I don't think it should be included in the "Versioned Default Values" section either.

hachi8833 committed:
Update guides/source/configuring.md

Thank you for the suggestion! Makes sense.

Co-authored-by: Hartley McGuire <skipkayhil@gmail.com>

hachi8833 committed:
Add linefeed

zzak commented:
> Since this configuration is not part of load_defaults, it should not include the table in the description, and I don't think it should be included in the "Versioned Default Values" section either.

@skipkayhil Maybe we can make that clearer, instead of  "Default Values for Target Version 7.0" as each heading just use `config.load_defaults(7.0)`. Or a combination so that it's obvious looking at this list 100 lines down what it's for.

intrip commented:
SQLite is becoming a more capable database and there are more companies using it in production. It is really good for self-hosted/small environments where you don't want to boot a separate server for the DB. I've noticed that there was also some recent work on improving the SQLite integration in Rails. I propose to consider the removal of this warning from Rails, doing it would also remove the need to handle Rails defaults.
/cc @dhh

dhh commented:
Yes, I think we should do away with this warning. Replace it with a notice in the docs/guide explaining the limitations of SQLite, as well as the advantages.

hachi8833 commented:
I close this PR and will recreate as a new one for 7-0/7-1-stable branches.
